zepl/contrib/clojure.vim

```diff
@@ -0,0 +1,67 @@
+" Description:  Text formatter for Clojure code.
+" File:         zepl/contrib/clojure.vim
+" Author:       Paul Fernandez <paul4nandez@gmail.com>
+" Help:         :help zepl-clojure
+" Legal:        No rights reserved.  Public domain.
+
+function s:stripLeadingEmptyLines(lines) abort
+   while !empty(a:lines)
+      if a:lines[0] =~# '\m^\s*$'
+         call remove(a:lines, 0)
+      else
+         break
+      endif
+   endwhile
+   return a:lines
+endfunction
+
+function s:stripTrailingEmptyLines(lines) abort
+   let idx = len(a:lines) - 1
+   while idx >= 0
+      if a:lines[idx] =~# '\m^\s*$'
+         call remove(a:lines, idx)
+         let idx -= 1
+      else
+         break
+      endif
+   endwhile
+   return a:lines
+endfunction
+
+function s:normalizeIndents(lines) abort
+   let depth = len(matchstr(a:lines[0], '\m\C^\s*'))
+   return map(a:lines, 'v:val[' . depth . ':]')
+endfunction
+
+function s:processLine(...) abort
+   " Remove trailing newlines.
+   let line = trim(a:2, "\r\n", 2)
```

Comment:
This is kind of personal, but it shouldn't hurt anything and may actually help others. I  have a function in my Vim config containing the following:

```vim
let l:ns = s:getClojureNS()
if strlen(l:ns)
  let l:cmd = "(do (require '" . l:ns . ")\n      (in-ns '" . l:ns . "))"
  execute 'ReplSend' . l:cmd
endif
```

I include the `\n` to avoid jamming both commands into a single line, but it comes out with an extra newline:

<img width="168" alt="Screen Shot 2022-05-11 at 4 34 53 PM" src="https://user-images.githubusercontent.com/458926/167968569-7c98baaa-ad05-44a7-ae81-32eb1bbc41f3.png">


This extra trim fixes it:

<img width="163" alt="Screen Shot 2022-05-11 at 4 36 10 PM" src="https://user-images.githubusercontent.com/458926/167968429-65b3a758-182d-4e8d-8ec2-ed0467f5805c.png">

Reply:
This should just work by using `\<CR>` instead of `\n`.

```vim
let l:ns = s:getClojureNS()
if strlen(l:ns)
  let l:cmd = "(do (require '" . l:ns . ")\<CR>      (in-ns '" . l:ns . "))"
  execute 'ReplSend' . l:cmd
endif
```

(Vim's handling of newlines is really confusing.  I don't understand it fully, but I've found that the rule of thumb is to try using `<CR>` first.  Or `\<CR>` if inside a `"` delimited string.)

Reply:
After echoing the line with `echomsg a:2`, I can see that an `^M` has been inserted. Going to do more experimenting to see where that's coming from. `^M` is a DOS line break but I'm on a Mac (see my other comment).

Reply:
It seems that there is `<LF>` and `<NL>` too.  I've only ever used `<CR>` which seems to have worked on Linux, Mac and Windows for me.